### PR TITLE
fix crash in ConnectorViewModel.cs

### PR DESCRIPTION
**Pull request by pinzart90**
2 regressions since https://github.com/DynamoDS/Dynamo/pull/12530

WpfVisualizationTests.HelixImageComparisonTests.HelixWatch3DViewModelTests.ColorCache_Updated_OnNode_Removed
WpfVisualizationTests.HelixWatch3DViewModelTests.ColorCache_Updated_OnNode_Removed

### CODE REVIEW

**Comment by pinzart90** [comment]
Passed here https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/5195/

File: src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs [hunk 1]
```diff
@@ -1127,9 +1135,14 @@ public override void Dispose()
 
             workspaceViewModel.DynamoViewModel.PropertyChanged -= DynamoViewModel_PropertyChanged;
             workspaceViewModel.DynamoViewModel.Model.PreferenceSettings.PropertyChanged -= DynamoViewModel_PropertyChanged;
-            Nodevm.PropertyChanged -= nodeViewModel_PropertyChanged;
-            NodeEnd.PropertyChanged -= nodeEndViewModel_PropertyChanged;
-            ConnectorPinViewCollection.CollectionChanged -= HandleCollectionChanged;
```
Comment (aparajit-pratap) [inline src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs:1132]: Why did you remove line 1132?

**Review by QilongTang — APPROVED** [review]
